### PR TITLE
Dc/feature/add bio UI

### PR DESCRIPTION
**Pull request by orangeboy55555**
Merge branch to develop

### CODE REVIEW

**Comment by ShahzaibParacha** [comment]
@orangeboy55555 and @WeiyuSun you will have to resolve the backend conflicts on this branch here

**Review by ShahzaibParacha — DISMISSED** [review]
Front-end changes look good to me, someone should review the backend stuff and we should be good to merge

**Review by DaTanUmanitoba — DISMISSED** [review]
The backend changes also look good to merge.

**Review by DaTanUmanitoba — DISMISSED** [review]
Have reviewed all the 18 changed files, they seem all good to merge.

**Comment by orangeboy55555** [comment]
Looks good

**Review by ShahzaibParacha — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -69,7 +69,7 @@ typings/
 .yarn-integrity
 
 # dotenv environment variables file
-.env
+backend/.env
```
Comment (ShahzaibParacha) [inline .gitignore:72]: lets keep a .gitignore in the backend for the backend, i already have one int he frontend

File: backend/middleware/user-middleware.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const Result = require("../util/Result");
 function userIdValidation(req, res, next) {
     console.log(`validating user_id`)
     let userId = req.query.user_id
```
Comment (ShahzaibParacha) [inline backend/middleware/user-middleware.js:7]: should probably replace these with a `const` if these are not changing

File: frontend/src/components/home/Home.jsx [hunk 3]
```diff
@@ -1,30 +1,66 @@
...
+import { useNavigate, Link } from "react-router-dom";
+import useAuthContext from "../../hooks/useAuthContext";
+import WidePost from "../widepost/WidePost";
 
 function Home() {
+  const [session, setSession] = useState(null);
+  const { userId, token, dispatch } = useAuthContext();
+
+  useEffect(() => {
+    const currentSession = {
```
Comment (orangeboy55555) [inline frontend/src/components/home/Home.jsx:11]: Might not need this

**Review by ShahzaibParacha — APPROVED** [review]
changes look good, we do need to have some discussions later

File: .gitignore (hunk 1 above)
Comment (WeiyuSun) [reply .gitignore:72]: so, if the root .env file is exactly what the backend needs. I just copy it to backend file

File: backend/middleware/user-middleware.js (hunk 2 above)
Comment (WeiyuSun) [reply backend/middleware/user-middleware.js:7]: sounds good, It is not a big problem. I prefer to upload it and backend env when we start sprint 3.is that ok?

**Comment by WeiyuSun** [comment]
Looks good for me

**Review by WeiyuSun — COMMENTED** [review]
looks good for me

**Review by DaTanUmanitoba — APPROVED** [review]
Seems good to merge. Well done!

**Pull request merged**